### DISCUSSION TITLE
How does flux wait for deployment to complete before applying the next kustomization?

### DISCUSSION THREAD

**Discussion by ylbeethoven** [fluxcd/flux2#4193] [Q&A] [answered]
I want to instruct flux to make sure service mesh deployment is completed before installing anything else. I installed service mesh (kuma) with helm.

I tried `.spec.dependsOn` ,  `.spec.healthChecks`, `.spec.wait` and `.spec.timeout` and none of these works as what I want. 

This means one of my other deployments pod does not have the sidecar injected because the service mesh control plane is not ready when it is deployed.

Does anyone know how I can achieve this?

Thank you.

**Comment by stefanprodan** [comment]
Follow this repository structure https://github.com/fluxcd/flux2-kustomize-helm-example:
- Place the Kuma HelmRelease in https://github.com/fluxcd/flux2-kustomize-helm-example/tree/main/infrastructure/controllers
- Place any Kuma custom resources in https://github.com/fluxcd/flux2-kustomize-helm-example/tree/main/infrastructure/configs
- Apps go in here https://github.com/fluxcd/flux2-kustomize-helm-example/tree/main/apps

Flux will first install Kuma, wait for it to become ready and only then proceed to Kuma CRs and apps.

**Reply by ylbeethoven** [reply]
Thanks @stefanprodan for the reply.

The issue I has was that the ingress controller (Kong) is sitting inside the same controller folder with Kuma.

I wanted to use KIC as the entry point of my service mesh so there must be a sidecar for Kong pods.

I guess the only solution is the put the KIC as an app in the app folder instead of putting it in controller folder. Is that right?

**Reply by stefanprodan** [reply, marked as answer]
Move KIC to `configs` or create a 3-layer structure in `infrastructure`.

**Reply by ylbeethoven** [reply]
Thanks @stefanprodan 

Adding extra layer works.
